File: binding/include/opengl/libqopenglversionfunctionsfactory.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qopenglversionfunctionsfactory.html
pub const qopenglversionfunctionsfactory = struct {

    /// New constructs a new QOpenGLVersionFunctionsFactory object.
    pub fn New(other: ?*anyopaque) QtC.QOpenGLVersionFunctionsFactory {
        return qtc.QOpenGLVersionFunctionsFactory_new(@ptrCast(other));
    }


    /// New2 constructs a new QOpenGLVersionFunctionsFactory object.
    pub fn New2(other: ?*anyopaque) QtC.QOpenGLVersionFunctionsFactory {
        return qtc.QOpenGLVersionFunctionsFactory_new2(@ptrCast(other));
    }


    /// New3 constructs a new QOpenGLVersionFunctionsFactory object.
    pub fn New3() QtC.QOpenGLVersionFunctionsFactory {
        return qtc.QOpenGLVersionFunctionsFactory_new3();
    }


    /// New4 constructs a new QOpenGLVersionFunctionsFactory object.
    pub fn New4(param1: ?*anyopaque) QtC.QOpenGLVersionFunctionsFactory {
        return qtc.QOpenGLVersionFunctionsFactory_new4(@ptrCast(param1));
    }

    /// CopyAssign shallow copies `other` into `self`.
    pub fn CopyAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QOpenGLVersionFunctionsFactory_CopyAssign(@ptrCast(self), @ptrCast(other));
    }

    /// MoveAssign moves `other` into `self` and invalidates `other`.
    pub fn MoveAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QOpenGLVersionFunctionsFactory_MoveAssign(@ptrCast(self), @ptrCast(other));
    }


    pub fn Get() QtC.QAbstractOpenGLFunctions {
        return qtc.QOpenGLVersionFunctionsFactory_Get();
    }

    pub fn OperatorAssign(self: ?*anyopaque, param1: ?*anyopaque) void {
        qtc.QOpenGLVersionFunctionsFactory_OperatorAssign(@ptrCast(self), @ptrCast(param1));
    }

    pub fn Get1(versionProfile: ?*anyopaque) QtC.QAbstractOpenGLFunctions {
        return qtc.QOpenGLVersionFunctionsFactory_Get1(@ptrCast(versionProfile));
    }

    pub fn Get2(versionProfile: ?*anyopaque, context: ?*anyopaque) QtC.QAbstractOpenGLFunctions {
        return qtc.QOpenGLVersionFunctionsFactory_Get2(@ptrCast(versionProfile), @ptrCast(context));
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QOpenGLVersionFunctionsFactory_Delete(@ptrCast(self));
    }
};

File: binding/src/opengl/libqopenglversionfunctionsfactory.cpp
#include <QOpenGLContext>
#include <QOpenGLVersionFunctionsFactory>
#include <QOpenGLVersionProfile>
#include <qopenglversionfunctionsfactory.h>
#include "libqopenglversionfunctionsfactory.h"
#include "libqopenglversionfunctionsfactory.hxx"

QOpenGLVersionFunctionsFactory* QOpenGLVersionFunctionsFactory_new(const QOpenGLVersionFunctionsFactory* other) {
	 return new QOpenGLVersionFunctionsFactory(*other);
}

QOpenGLVersionFunctionsFactory* QOpenGLVersionFunctionsFactory_new2(QOpenGLVersionFunctionsFactory* other) {
	 return new QOpenGLVersionFunctionsFactory(*other);
}

QOpenGLVersionFunctionsFactory* QOpenGLVersionFunctionsFactory_new3() {
	 return new QOpenGLVersionFunctionsFactory();
}

QOpenGLVersionFunctionsFactory* QOpenGLVersionFunctionsFactory_new4(const QOpenGLVersionFunctionsFactory* param1) {
	 return new QOpenGLVersionFunctionsFactory(*param1);
}

void QOpenGLVersionFunctionsFactory_CopyAssign(QOpenGLVersionFunctionsFactory* self, QOpenGLVersionFunctionsFactory* other) {
    *self = *other;
}

void QOpenGLVersionFunctionsFactory_MoveAssign(QOpenGLVersionFunctionsFactory* self, QOpenGLVersionFunctionsFactory* other) {
    *self = std::move(*other);
}

QAbstractOpenGLFunctions* QOpenGLVersionFunctionsFactory_Get() {
	return QOpenGLVersionFunctionsFactory::get();
}

void QOpenGLVersionFunctionsFactory_OperatorAssign(QOpenGLVersionFunctionsFactory* self, const QOpenGLVersionFunctionsFactory* param1) {
	self->operator=(*param1);
}

QAbstractOpenGLFunctions* QOpenGLVersionFunctionsFactory_Get1(const QOpenGLVersionProfile* versionProfile) {
	return QOpenGLVersionFunctionsFactory::get(*versionProfile);
}

QAbstractOpenGLFunctions* QOpenGLVersionFunctionsFactory_Get2(const QOpenGLVersionProfile* versionProfile, QOpenGLContext* context) {
	return QOpenGLVersionFunctionsFactory::get(*versionProfile, context);
}

void QOpenGLVersionFunctionsFactory_Delete(QOpenGLVersionFunctionsFactory* self) {
    delete self;
}

